LibGfx: Only include frames with fcTL chunks in the animation

aplefull opened the pull request:
Fixes #4519.
As Sam mentioned in the issue, we were including base IDAT data in the animation frames. Now we only include frames with explicit fcTL chunks in the animation.
We also fall back to the IDAT data when APNG has acTL but no fcTL chunks. It's a weird edge case, but that's how Chrome and Firefox handle such files and now we match them.

LucasChollet commented on Libraries/LibGfx/ImageFormats/PNGLoader.cpp:
```diff
@@ -334,6 +339,15 @@ ErrorOr<size_t> PNGLoadingContext::read_frames(png_structp png_ptr, png_infop in
             }
         }
+
+        this->frame_count = animation_frame_count;
```
We generally don't use the `this->` prefix to access members.

LucasChollet commented on Tests/LibGfx/TestImageDecoder.cpp:
```diff
@@ -363,6 +363,22 @@ TEST_CASE(test_png)
 
+TEST_CASE(test_apng)
+{
+    auto file = TRY_OR_FAIL(Core::MappedFile::map(TEST_INPUT("png/apng-1-frame.png"sv)));
```
I suppose this is `062.png` from `fDAT-inherits-cICP.html`, it would be nice to say where you got this file in the commit message.

LucasChollet reviewed (changes requested):
Thanks for looking into this, and congrats on finding the root cause.
For the contribution though, let's get into the habit of putting relevant information in the commit itself rather than the PR, so it appears in the git log.

Also, your fix isn't exactly doing the right thing, as per this spec quote (from the third edition):
> The static image may be included as the first frame of the animation by the presence of a single fcTL chunk before IDAT. Otherwise, the static image is not part of the animation.

aplefull commented:
> Thanks for looking into this, and congrats on finding the root cause. For the contribution though, let's get into the habit of putting relevant information in the commit itself rather than the PR, so it appears in the git log.
> 
> Also, your fix isn't exactly doing the right thing, as per this spec quote (from the third edition):
> 
> > The static image may be included as the first frame of the animation by the presence of a single fcTL chunk before IDAT. Otherwise, the static image is not part of the animation.

I removed `this->` and updated commit message. But I'm not sure I understand your comment correctly. I'm aware of this APNG feature and my fix accounts for this. IDAT is included in animation if there is a valid fcTL before it. I just made a test file for this (fcTL → IDAT → fcTL → fdAT) and it works as expected. Attaching here just in case.

![animated_test](https://github.com/user-attachments/assets/0088f58c-7643-4392-b76f-075385937dcc)

Could you please explain what you meant by this comment?

LucasChollet commented on Libraries/LibGfx/ImageFormats/PNGLoader.cpp:
```diff
@@ -334,6 +339,15 @@ ErrorOr<size_t> PNGLoadingContext::read_frames(png_structp png_ptr, png_infop in
+
+        frame_count = animation_frame_count;
+
+        // If we didn't find any valid animation frames with fcTL chunks, fall back to using the base IDAT data as a single frame.
```
This comment confused me, I thought that it would be the only path where an `IDAT` chunk's data could be included in `frame_descriptors`.

aplefull replied:
Oh, right, I shoud've been more clear. It's for the case like this: IHDR → acTL → IDAT → IEND. So techlically it's APNG, but without any animation frames. Other browsers just display base IDAT image in this case, so I added this check

aplefull replied:
Added the spec quote 🫡

LucasChollet reviewed (commented):
Oops, I wrote the comment yesterday, but I also forgot to submit it.
